File: Frontend/Tasksphere/src/features/auth/services/authService.ts
import { axiosClient } from "../../../api/axiosClient"
import type { LoginRequest, LoginResponse, RegisterRequest, RegisterResponse } from "../types/auth.types"

export const login = async (data: LoginRequest, signal?: AbortSignal): Promise<LoginResponse> => {
    const response = await axiosClient.post("/auth/login", data, { signal });
    return response.data;
}

export const registerUser = async (data: RegisterRequest, signal?: AbortSignal): Promise<RegisterResponse> => {
    const response = await axiosClient.post("/auth/register", data, { signal });
    return response.data;
}

File: Backend/TaskSphere/src/main/java/com/prathamesh/tasksphere/service/AuthService.java
package com.prathamesh.tasksphere.service;

import com.prathamesh.tasksphere.dto.LoginRequest;
import com.prathamesh.tasksphere.dto.LoginResponse;
import com.prathamesh.tasksphere.dto.RegisterRequest;
import com.prathamesh.tasksphere.dto.RegisterResponse;

public interface AuthService {
	RegisterResponse register(RegisterRequest request);
	LoginResponse login(LoginRequest request);
}
